packages/agent/src/test-managed-agent.ts

```diff
@@ -0,0 +1,243 @@
+      didResolver  : didResolver,
+      messageStore : dwnMessageStore
+    });
+
+    // Instantiate a DwnManager using the custom DWN instance.
+    const dwnManager = new DwnManager({ dwn });
+
+    // Instantiate an RPC Client.
+    const rpcClient = new Web5RpcClient();
+
+    const agent = new agentClass({
+      agentDid: '',
+      appData,
+      didManager,
+      didResolver,
+      dwnManager,
+      identityManager,
+      keyManager,
+      rpcClient,
+    });
+
+    return new TestManagedAgent({
+      agent,
+      agentStores,
+      appDataStore,
+      dwn,
+      dwnDataStore,
+      dwnEventLog,
+      dwnMessageStore,
+    });
+  }
+
+  async createAgentDid(): Promise<void> {
+    // Create an a DID and key set for the Agent.
+    const agentDid = await DidKeyMethod.create({ keyAlgorithm: 'Ed25519' });
+    const privateCryptoKey = await Jose.jwkToCryptoKey({ key: agentDid.keySet.verificationMethodKeys![0].privateKeyJwk! });
+    const publicCryptoKey = await Jose.jwkToCryptoKey({ key: agentDid.keySet.verificationMethodKeys![0].publicKeyJwk! });
+    const agentSigningKey = { privateKey: privateCryptoKey, publicKey: publicCryptoKey };
+
+    // Set the DID as the default signing key.
+    const alias = await this.agent.didManager.getDefaultSigningKey({ did: agentDid.did });
+    const defaultSigningKey = cryptoToPortableKeyPair({ cryptoKeyPair: agentSigningKey, keyData: { alias, kms: 'memory' } });
+    await this.agent.keyManager.setDefaultSigningKey({ key: defaultSigningKey });
+
+    // Set the DID as the Agent's DID.
+    this.agent.agentDid = agentDid.did;
+  }
+
+  private static useDiskStorage(options: { agent?: Web5ManagedAgent, testDataLocation: string }) {
+    const { agent, testDataLocation } = options;
+    const testDataPath = (path: string) => `${testDataLocation}/${path}`;
+
+    const appDataStore = new LevelStore(testDataPath('APPDATA'));
+    const appData = new AppDataVault({
+      keyDerivationWorkFactor : 1,
+      store                   : appDataStore
+    });
+
+    const didManager = new DidManager({
+      agent,
+      didMethods : [DidIonMethod, DidKeyMethod],
+      store      : new DidStoreDwn()
+    });
+
+    const identityManager = new IdentityManager({
+      agent,
+      store: new IdentityStoreDwn()
+    });
+
+    const localKmsDwn = new LocalKms({
+      agent,
+      kmsName         : 'local',
+      keyStore        : new KeyStoreDwn({ schema: 'https://identity.foundation/schemas/web5/kms-key' }),
+      privateKeyStore : new PrivateKeyStoreDwn()
+    });
+    const localKmsMemory = new LocalKms({
+      agent,
+      kmsName: 'memory'
+    });
+    const keyManager = new KeyManager({
+      agent,
+      kms: {
+        local  : localKmsDwn,
+        memory : localKmsMemory
+      },
+      store: new KeyStoreDwn({ schema: 'https://identity.foundation/schemas/web5/managed-key' })
```

Comment:
this url doesn't exist, is that ok? 
should we store the schemas locally so that remote resolution doesn't have to happen ?

Reply:
Yes -- it is never actually resolved
simply used as a way to filter the queries
intent is to replace with a protocol